doc/algorithms.md

```diff
@@ -0,0 +1,727 @@
+# Core Algorithms used in MeetEval
+
+This file contains simplified implementations of the core algorithms used in the MeetEval toolkit.
+These implementations are meant for documentation.
+The actual implementations in the toolkit are more complex and track the alignment along with the distance computation, which is omitted here for clarity.
+
+All algorithms in this file operate on a _character level_ instead of the word level.
+This doesn't change the algorithms by themselves, but simplifies the implementations (because `str.split()` can be omitted).
+Keep in mind that these simplified implementations may not cover all edge-cases correctly.
+
+## Levenshtein Distance
+
+> [!NOTE]
+> 
+> [View C++ Implementation](../meeteval/wer/matching/levenshtein.h#L8)
+
+The Levenshtein distance (https://en.wikipedia.org/wiki/Levenshtein_distance, [the english translation of the original paper](https://nymity.ch/sybilhunting/pdf/Levenshtein1966a.pdf)) is the core of the word error rate.
+We use a version of the Wagner-Fisher algorithm that only uses linear memory (see, for example, https://www.baeldung.com/cs/levenshtein-distance-computation).
+There are other algorithms that are more efficient (.e.g., a bit-parallel algorithm in https://dl.acm.org/doi/10.1145/316542.316550), but we decide to stick to the simpler algorithms that are easier to understand and extend.
```

Comment:
There are other algorithms that are more efficient (.e.g., a bit-parallel algorithm in https://dl.acm.org/doi/10.1145/316542.316550), but their extension to the computational extensive algorithm is unknown.